### DISCUSSION TITLE
ImU32 Colors

### DISCUSSION THREAD

**Discussion by SlavicBlin** [tupoy-ya/TupoyeMenu#77] [Q&A] [answered]
How do you get the Color Code for something like the buttons?
By default its set to " 4293353517 " and i would like to turn in into purple by default but have no idea how. I have went online and alot of them said pack it but have not shown how to.
Maybe you could help

**Comment by tupoy-ya** [comment]
Check the code in `view_gui_settings.cpp` - all color pickers there use the
```cpp
ImGui::ColorConvertFloat4ToU32(col_gui);
```
function.

If you only want to set the default color then just set it in the settings normally and then copy/paste the value from `settings.json`.

**Reply by SlavicBlin** [reply]
Yes i got that part but then how do you know that the numbers " 4293353517 " are equal to blue? is there a convertor for RGB to U32?

**Reply by tupoy-ya** [reply, marked as answer]
> Is there a convertor for RGB to U32?

Yes `ImU32 ImGui::ColorConvertFloat4ToU32(const ImVec4& in)` it takes `ImVec4` which is basically R G B A mapped as x y z w, and returns `ImU32`.

There is also `ImVec4 ColorConvertU32ToFloat4(ImU32 in)` which does the opposite and converts `ImU32` into `ImVec4`.

If you are asking for a standalone program just for these functions then i don't think there is one.

**Comment by SlavicBlin** [comment]
> > Is there a convertor for RGB to U32?
> 
> Yes `ImU32 ImGui::ColorConvertFloat4ToU32(const ImVec4& in)` it takes `ImVec4` which is basically R G B A mapped as x y z w, and returns `ImU32`.
> 
> There is also `ImVec4 ColorConvertU32ToFloat4(ImU32 in)` which does the opposite and converts `ImU32` into `ImVec4`.
> 
> If you are asking for a standalone program just for these functions then i don't think there is one.

I see i see its seems very confusing and seems like it would be hard to get a specific RGB color to get converted into a imU32.
Anyways thank you very much for the help 👍
